File: IntroductoryProblems/04_increasingArray/04_increasingArray.py
#https://cses.fi/problemset/task/1094

def increasingArray(arr):
    count = 0
    for i in range (1,len(arr)):
        if(arr[i]<arr[i-1]):
            count+= (arr[i-1]-arr[i])
    return count
    
    
arr = [3,2,5,1,7]
print(increasingArray(arr))

File: IntroductoryProblems/04_increasingArray/04_increasingArray.java
//https://cses.fi/problemset/task/1094
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

class IncreasingArray
{
    public static int minimumMove(ArrayList<Integer>arr)
    {
        int count = 0;

        for(int i=1;i<arr.size();++i)
        {
    
            if(arr.get(i-1)>arr.get(i))
                count += (arr.get(i-1)-arr.get(i));
        }

        return count;
    }
   public static void main(String[] args) 
   {
        Scanner sc = new Scanner(System.in);
        ArrayList<Integer>arr = new ArrayList<>(Arrays.asList(3,2,5,1,7));
        sc.close();
        System.out.println(minimumMove(arr));
    
   } 
}